### PR TITLE
Support Scala.js and further reorg (part1)

### PR DESCRIPTION
**Pull request by aslesarenko**
In this PR:
- [x] Scala.js compilation configured for common, core-lib and interpreter modules
- [x] consensus code further minimized
- [x] more code moved to compiler `sc` module
- [x] unused code removed

### CODE REVIEW

File: common/jvm/src/test/scala/scalan/util/FileUtil.scala [hunk 1]
```diff
@@ -6,10 +6,36 @@ import java.nio.file._
 import java.nio.file.attribute.BasicFileAttributes
 import scala.Console
 import scala.collection.JavaConverters._
-import scalan.util.StringUtil.StringUtilExtensions
+import scalan.util.StringUtil.{StringUtilExtensions, cleanFileName}
```
Comment (kushti) [inline common/jvm/src/test/scala/scalan/util/FileUtil.scala:9]: StringUtilExtensions import not used

import scala.Console    as well

File: interpreter/shared/src/main/scala/sigmastate/sigmastate.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import sigmastate.Values._
+import sigmastate.lang.CheckingSigmaBuilder
+
+package object sigmastate {
+  import CheckingSigmaBuilder._
+
+  /**
+    * SInt addition
```
Comment (kushti) [inline interpreter/shared/src/main/scala/sigmastate/sigmastate.scala:8]: Why SInt? What does this comment mean ?

**Review by kushti — APPROVED** [review]

File: common/jvm/src/test/scala/scalan/util/FileUtil.scala (hunk 1 above)
Comment (aslesarenko) [reply common/jvm/src/test/scala/scalan/util/FileUtil.scala:9]: fixed (except Console, which is used)

File: interpreter/shared/src/main/scala/sigmastate/sigmastate.scala (hunk 2 above)
Comment (aslesarenko) [reply interpreter/shared/src/main/scala/sigmastate/sigmastate.scala:8]: fixed

**Pull request merged**